src/actions/configuration/first-time-configuration-action.php

```diff
@@ -0,0 +1,300 @@
+			];
+		}
+		return (object) [
+			'success'  => false,
+			'status'   => 500,
+			'error'    => 'Could not save some options in the database',
+			'failures' => $failures,
+		];
+	}
+
+	/**
+	 * Stores the values for the social profiles.
+	 *
+	 * @param array $params The values to store.
+	 *
+	 * @return object The response object.
+	 */
+	public function set_person_social_profiles( $params ) {
+		$social_profiles = \array_filter(
+			$params,
+			function ( $key ) {
+				return $key !== 'user_id';
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+
+		$failures = $this->social_profiles_helper->set_person_social_profiles( $params['user_id'], $social_profiles );
+
+		if ( \count( $failures ) === 0 ) {
+			return (object) [
+				'success' => true,
+				'status'  => 200,
+			];
+		}
+		return (object) [
+			'success'  => false,
+			'status'   => 500,
+			'error'    => 'Could not save some options in the database',
+			'failures' => $failures,
+		];
+	}
+
+	/**
+	 * Gets the values for the social profiles.
+	 *
+	 * @param int $user_id the person id.
+	 *
+	 * @return object The response object.
+	 */
+	public function get_person_social_profiles( $user_id ) {
+
+		return (object) [
+			'success'         => true,
+			'status'          => 200,
+			'social_profiles' => $this->social_profiles_helper->get_person_social_profiles( $user_id ),
+		];
+	}
+
+	/**
+	 * Stores the values to enable/disable tracking.
+	 *
+	 * @param array $params The values to store.
+	 *
+	 * @return object The response object.
+	 */
+	public function set_enable_tracking( $params ) {
+		$success      = true;
+		$option_value = $this->options_helper->get( 'tracking' );
+
+		if ( $option_value !== $params['tracking'] ) {
+			$success = $this->options_helper->set( 'tracking', $params['tracking'] );
+		}
+
+		if ( $success ) {
+			return (object) [
+				'success' => true,
+				'status'  => 200,
+			];
+		}
+		return (object) [
+			'success' => false,
+			'status'  => 500,
+			'error'   => 'Could not save the option in the database',
+		];
+	}
+
+	/**
+	 * Checks if the current user has the capability a specific user.
+	 *
+	 * @param int $user_id The id of the user to be edited.
+	 *
+	 * @return object The response object.
+	 */
+	public function check_capability( $user_id ) {
+		if ( $this->social_profiles_helper->can_edit_profile( $user_id ) ) {
+			return (object) [
+				'success' => true,
+				'status'  => 200,
+			];
+		}
+
+		return (object) [
+			'success' => false,
+			'status'  => 403,
+		];
+	}
+
+	/**
+	 * Stores the first time configuration state.
+	 *
+	 * @param array $params The values to store.
+	 *
+	 * @return object The response object.
+	 */
+	public function save_configuration_state( $params ) {
+		$configuration_data                      = [];
+		$configuration_data['finishedSteps']     = $params['finishedSteps'];
+		$configuration_data['indexablesBySteps'] = $params['indexablesBySteps'];
+		$configuration_data['priority']          = $params['priority'];
```

Comment:
I think that these need both `isset()` checks and also sanitization since we're saving them in the db

Reply:
We actually don't need indexables by step and also not priority